common/src/hash/cpp/Names.cpp

```diff
@@ -29,7 +29,7 @@ constexpr uint32_t PrintableHexSize(uint32_t input_length) {
 
 void PrintableHex(const uint8_t* data, uint32_t data_length, char* hex) {
   static const char* hex_digits = "0123456789ABCDEF";
-  for(int i = 0; i < data_length; ++i) {
+  for (uint32_t i = 0; i < data_length; ++i) {
```

Comment:
These for loops need to be improved indeed. Maybe a range-for would even be better here? Something like `for (auto i: iota(0, length)) { … }` where `iota` is something like [this](https://en.cppreference.com/w/cpp/ranges/iota_view) from C++20. Unfortunately the standard `iota` wouldn't do, because `iota` allows bounds to have different types and so the type of `i` would be `int`, but we can create our own simple iterator that would enforce bounds type equality.

At any rate, I suggest to postpone this refactoring a bit. At least with the current warning setup, we would get a compile time error if `data_length` becomes signed.